Cleaned Up intake and robot container

aaron345678 opened the pull request:
cleaned up intake methods and robot container index for arm

aaron345678 committed:
Cleaned Up intake and robot container

ProfessorAtomicManiac commented on src/main/java/org/carlmontrobotics/RobotContainer.java:
```diff
@@ -340,10 +340,14 @@ private void registerAutoCommands(){
-    NamedCommands.registerCommand("ArmToPodium", new ArmToPos(arm, Armc.PODIUM_ANGLE_RAD, 2));
-    NamedCommands.registerCommand("ArmToGround", new ArmToPos(arm, GROUND_INTAKE_POS, 1));
+    NamedCommands.registerCommand("ArmToAmp",
+        new ArmToPos(arm, Armc.AMP_ANGLE_RAD, AMP_RPM_INDEX));
```
Can you combine the position + index? Ideally you can just do `new ArmToPos(arm, pos)` and it should take care of the rest instead of having to add additional parameters.

ProfessorAtomicManiac commented on src/main/java/org/carlmontrobotics/Constants.java:
```diff
@@ -77,6 +77,9 @@ public static final class Effectorc {
 		public static final double PODIUM_RPM = 4000;
 		public static final double SPEAKER_RPM = 2100;
 		public static final double[] RPM_SELECTOR = {AMP_RPM, SUBWOOFER_RPM, PODIUM_RPM};
+		public static final int AMP_RPM_INDEX = 0;
+		public static final int SUBWOOFER_RPM_INDEX = 1;
+		public static final int PODIUM_RPM_INDEX = 2;
```
Adding on to my comment above, but IMO its really annoying to have to add an index everytime we want a new arm position. This feels like something only the person who originally wrote the code would know to do.

ProfessorAtomicManiac reviewed (changes requested):
Please look at:
methods that use outdated components:
![image](https://github.com/DeepBlueRobotics/RobotCode2024/assets/59379639/de9c2adc-2735-4669-8565-a987c094953a)
![image](https://github.com/DeepBlueRobotics/RobotCode2024/assets/59379639/936b8ce4-d360-4483-870a-c9751bff9a5b)


methods not used anymore:
![image](https://github.com/DeepBlueRobotics/RobotCode2024/assets/59379639/619ba374-ef4e-41c9-88d2-a09f18723e76)

In `RobotContainer.java` you have the following methods:
- `setBindingsManipulatorENDEFF()`
- `setBindingsManipulatorNEO()`

This is not exhaustive but you get the idea

DriverStationComputer committed:
Cleaned up soe comands and constants for robot container

brettle commented:
Merging master into this branch will probably fix the current build failure.

ProfessorAtomicManiac commented on src/main/java/org/carlmontrobotics/RobotContainer.java:
```diff
@@ -221,48 +221,17 @@ private void setBindingsManipulatorNEO() {
-    new JoystickButton(manipulatorController, Button.kRightStick.value).whileTrue(new PassToIntake(intakeShooter));
-    new JoystickButton(manipulatorController, Button.kX.value).onTrue(new ArmToPos(arm, SPEAKER_ANGLE_RAD,1));
-    //TODO: test angles for pov button BEFORE climbing
+    new JoystickButton(manipulatorController, PASS_TO_OUTTAKE_STICK)
+        .onTrue(new PassToOuttake(intakeShooter));
+    new JoystickButton(manipulatorController, PASS_TO_INTAKE_STICK)
+        .whileTrue(new PassToIntake(intakeShooter));
```
Is there a reason why `PassToOuttake` uses a timer and `PassToIntake` uses distance sensors as their end conditions?

aaron345678 replied:
Was added a LONG time ago because we were overriding the current limit. Has now been changed to just distance sensors

ProfessorAtomicManiac commented on src/main/java/org/carlmontrobotics/RobotContainer.java:
```diff
@@ -221,48 +221,17 @@ private void setBindingsManipulatorNEO() {
+        .onTrue(new PassToOuttake(intakeShooter));
+    new JoystickButton(manipulatorController, PASS_TO_INTAKE_STICK)
+        .whileTrue(new PassToIntake(intakeShooter));
+    new JoystickButton(manipulatorController, Button.kX.value)
```
Please put raw button values in `Constants.java`

ProfessorAtomicManiac replied:
~there are still some raw button values present~

ProfessorAtomicManiac commented on src/main/java/org/carlmontrobotics/RobotContainer.java:
```diff
@@ -221,48 +221,17 @@ private void setBindingsManipulatorNEO() {
+        .onTrue(new ArmToPos(arm, SPEAKER_ANGLE_RAD, 1));
     new POVButton(manipulatorController, 0).onTrue(new ArmToPos(arm, CLIMB_POS, 0));
     new POVButton(manipulatorController, 180).onTrue(new Climb(arm));
     new POVButton(manipulatorController, 270).onTrue(new ArmToPos(arm, PODIUM_ANGLE_RAD, 2));
```
Please put raw POVButton values in `Constants.java`

ProfessorAtomicManiac reviewed (changes requested):
- [x] Some button mappings in `RobotContainer.java` use raw button values, for example on line 222. Please move them to `Constants.java`.
```java
    // Move "Button.kY.value" to the "Constants.OI.Manipulator" class
    new JoystickButton(manipulatorController, Button.kY.value).onTrue(...)
```
- [x] Files/methods like "IntakeNEO.java" should be renamed to "Intake.java" since we don't care about whether we use NEO or not. I would also recommend renaming "Eject.java" to "EjectIntakeSide.java" so people can more easily differentiate it from "EjectOuttakeSide.java"
- [x] Did you decide to keep `IntakeTesting.java` or you forgot about it? If you want to keep it, maybe add some comments so people don't delete later on since it's not used.
- [x] To combine the position and index parameter for `ArmToPos` method, I would create an class or enum which stores the pos and index variable. Then you can just call `new ArmToPos(arm, [insert new class/enum].PODIUM_ANGLE_RAD)`. For two examples, please see [`RunRoller.java`](https://github.com/DeepBlueRobotics/RobotCode2023/blob/master/src/main/java/org/carlmontrobotics/robotcode2023/commands/RunRoller.java) and [`SetArmWristGoalPreset.java`](https://github.com/DeepBlueRobotics/RobotCode2023/blob/master/src/main/java/org/carlmontrobotics/robotcode2023/commands/SetArmWristGoalPreset.java) and how they interact with `Constants.java` and their respective subsystems in `RobotCode2023.java`.

A few tips:
- If you highlight something + right click -> "Rename Symbol", you can edit the variable/method name and you wouldn't have to manually go edit every single other instance where it is used.
~- `Alt` + `Shift` + `F` auto-formats your files~ (don't use these see dean's comment)
- `Alt` + `Shift` + `O` removes unused imports and organizes them. Beware that it does import static individual constants, so you should change it back to `*`.
~- `Ctrl` + `Shift` + `P` then type `Trim Trailing Spaces`~ (don't use these see dean's comment)

brettle commented:
> * `Alt` + `Shift` + `F` auto-formats your files
> 
> * `Ctrl` + `Shift` + `P` then type `Trim Trailing Spaces`

Those 2 things should now happen automatically on any changed lines in the file when the file is saved. I generally recommend *against* reformatting an entire file (unless it is being done as a separate formatting-only PR) because it can make it hard to see "real" changes when doing reviews.

aaron345678 committed:
Cleaned up most of robot container buttons. Removed switch rpm index. Changed intake name. Still not done

FriedLongJohns commented:
_Looks_ fine to me, but I'm failing the lib199 gradle import so I can't build it.

ProfessorAtomicManiac commented on src/main/java/org/carlmontrobotics/Constants.java:
```diff
@@ -447,10 +446,17 @@ public static final class Manipulator {
 
 		public static final double JOY_THRESH = 0.01;
 		public static final double MIN_AXIS_TRIGGER_VALUE = 0.2;// woah, this is high.
+		public static final int UP_D_PAD = 0;
+		public static final int DOWN_D_PAD = 180;
+		public static final int LEFT_D_PAD = 270;
+		public static final int A_BUTTON = Button.kA.value;
```
Typically, we name such constants with what command they are associated with. For example,
`public static final int RAISE_CLIMBER = 0; // Up D-pad`
btw your `RAISE_CLIMBER` variable on line 447 isn't actually used anywhere, so please double check that the button mappings are all updated.

brettle replied:
I find @aaron345678's approach much more useful. It means that if I want to know which button runs a particular command I can just look for where it is referenced and I immediately know. With @ProfessorAtomicManiac's approach I additionally need to find the button constant in Constants to see what the comment says. I don't see what benefit that provides.

ProfessorAtomicManiac replied:
My thinking was that I can go to `Constants.java`, and see all the buttons. I can see which button (what the variable is assigned) runs a particular command (which is in the variable name). This is what we've done in the past.
Correct me if I'm wrong but using Aaron's way would mean I can go to `Constants.java`, find that button then can use VSCode features to find where that button is referenced in the code to know what the button runs. Or I can go to `RobotContainer.java`, and Ctrl+F to find where that button is used.

Tbh, I don't care which way we use and that should really be left up to @FriedLongJohns.

ProfessorAtomicManiac replied:
@aaron345678 you can ignore my comments about renaming UP_D_PAD or my other comment about how there are "raw buttons present" for now. The important thing that I want you to do is to get rid of unused constants like `RAISE_CLIMBER` so people don't try to change that variable thinking it changes the button to raise the climber.

ProfessorAtomicManiac commented on src/main/java/org/carlmontrobotics/commands/SwitchRPMShootNEO.java:
```diff
@@ -11,22 +14,35 @@
         this.intakeShooter = intakeShooter;
-        addRequirements(intakeShooter);
+        this.arm = arm;
+        addRequirements(intakeShooter, arm);
```
if you aren't actually controlling the arm but only obtaining values from the arm, you don't need to set it as a requirement. Otherwise, you wouldn't be able to run arm commands while this is running.

aaron345678 committed:
Cleaned up robot container buttons and renamed SwitchRPM cmd to outtake. Removed Arm req from outtake

aaron345678 committed:
Deleted intake testing

ProfessorAtomicManiac commented on src/main/java/org/carlmontrobotics/commands/Outtake.java:
```diff
@@ -9,39 +12,56 @@
         intakeShooter.setMaxOuttake(1);
-        timer.reset();
+        currentPos = arm.getArmPos();
+        if (currentPos >= PODIUM_ANGLE_RAD + 0.1
+                && currentPos <= PODIUM_ANGLE_RAD - 0.1) {
+            rpmAmount = PODIUM_RPM;
+        } else if (currentPos >= SPEAKER_ANGLE_RAD - 0.1) {
+            rpmAmount = SPEAKER_RPM;
+        } else {
+            rpmAmount = SUBWOOFER_RPM;
+        }
```
shouldn't it be currentPos <= PODIUM_ANGLE_RAD + 0.1 instead of >=? Also you could just do `Math.abs(currentPos - PODIUM_ANGLE_RAD) <= 0.1` for simplicity.

FriedLongJohns replied:
unimportant.

ProfessorAtomicManiac commented on src/main/java/org/carlmontrobotics/Constants.java:
```diff
@@ -437,20 +436,25 @@ public static final class Manipulator {
 			public static final Axis SHOOTER_BUTTON = Axis.kRightTrigger;
 			public static final int EJECT_BUTTON = Button.kLeftBumper.value;
-			public static final int AMP_BUTTON = Button.kRightBumper.value;
+			public static final int OPPISITE_EJECT = Button.kRightBumper.value;
```
typo

FriedLongJohns committed:
bump gradlew, fix typo, fix err

hardcoded NEO_BUILTIN_ENCODER_CPR since it was causing an issue since it doesn't exist

ProfessorAtomicManiac committed:
removed useless bindings

ProfessorAtomicManiac commented:
please merge from master

aaron345678 committed:
Added TestRPM command for 6/30 testing

ProfessorAtomicManiac committed:
Merge branch 'master' of https://github.com/DeepBlueRobotics/RobotCode2024 into cleanerCode

ProfessorAtomicManiac committed:
other small fixes

ProfessorAtomicManiac reviewed (approved):
hopefully merge didn't mess anything up

ProfessorAtomicManiac commented:
fixed #76
